src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -839,6 +839,18 @@ case class ApplyMethod(posn: Position, lhs: AST, method: String, args: Array[AST
           parseError(s"method `$method' expects a lambda function [param => Boolean], got [param => ${body.`type`}]")
         `type` = TBoolean
 
+      case (arr: TArray, "sortBy", rhs) =>
+        lhs.typecheck(ec)
+        val (param, body) = rhs match {
+          case Array(Lambda(_, p, b)) => (p, b)
+          case _ => parseError(s"method `$method' expects a lambda function [param => T] with T of string or numeric type, " +
+            s"e.g. `x => x.length'") // FIXME: better examples
```

Comment:
Without context (what is x?) this is more confusing than not.  I don't think error messages are the place for examples.